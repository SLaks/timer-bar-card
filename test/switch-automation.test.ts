import { HomeAssistant, PlaywrightBrowser, PlaywrightElement } from "hass-taste-test";
import { toMatchDualSnapshot, waitForTimerTime } from "./util";

expect.extend({ toMatchDualSnapshot });

/** Home Assistant Configuration */
const CONFIGURATION_YAML = `
input_boolean:
  switch:

input_number:
  duration:
    min: 1
    max: 20
    initial: 4

automation switch_off:
  alias: 'Turn switch off after some time, configurable with an input_number'
  trigger:
    - platform: state
      entity_id: input_boolean.switch
      to: 'on'
  condition: []
  action:
    - delay: "{{ states('input_number.duration') }}"
    - service: input_boolean.turn_off
      target:
        entity_id: input_boolean.switch
`;

/** Timer Bar Card Configuration */
const CONFIGURATION_CARD = {
  type: "custom:timer-bar-card",
  entities: ["input_boolean.switch"],
  duration: { entity: "input_number.duration", "units": "seconds" },
}

let hass: HomeAssistant<PlaywrightElement>;

beforeAll(async () => {
  hass = await HomeAssistant.create(CONFIGURATION_YAML, {
    browser: new PlaywrightBrowser(process.env.BROWSER || "firefox"),
  });
  await hass.addResource(__dirname + "/../dist/timer-bar-card.js", "module");
}, 30000);
afterAll(async () => await hass.close());

it("Switch with input_number turns off", async () => {
  const dashboard = await hass.Dashboard([CONFIGURATION_CARD]);
  const card = dashboard.cards[0];
  await hass.callService('homeassistant', 'turn_on', {}, { entity_id: "input_boolean.switch" });
  await waitForTimerTime(card, "00:00:02");
  await expect(card).toMatchDualSnapshot("running");

  // wait for timer to end; by this time automation should turn the switch off
  await waitForTimerTime(card, "00:00:01");
  await new Promise(r => setTimeout(r, 2000));

  await expect(card).toMatchDualSnapshot("idle");
}, 9000);
