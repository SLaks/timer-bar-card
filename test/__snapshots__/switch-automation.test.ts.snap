// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch with input_number turns off: idle 1`] = `
"<timer-bar-card hass=\\"[object Object]\\">
  <ha-card outlined=\\"false\\">
    <slot></slot>
    <div class=\\"card-content\\">
      <timer-bar-entity-row config=\\"[object Object]\\" hass=\\"[object Object]\\">
        <hui-generic-entity-row config=\\"[object Object]\\" no-secondary=\\"\\">
          <state-badge _showIcon=\\"true\\" class=\\"pointer\\" stateObj=\\"[object Object]\\" tabindex=\\"0\\">
            <ha-state-icon data-state=\\"off\\">…</ha-state-icon>
          </state-badge>
          <div class=\\"info pointer text-content\\" title=\\"switch\\">
            switch
          </div>
          <slot></slot>
          <div class=\\"text-content\\">
            Off
          </div>
        </hui-generic-entity-row>
      </timer-bar-entity-row>
    </div>
  </ha-card>
</timer-bar-card>
"
`;

exports[`Switch with input_number turns off: running 1`] = `
"<timer-bar-card hass=\\"[object Object]\\">
  <ha-card outlined=\\"false\\">
    <slot></slot>
    <div class=\\"card-content\\">
      <timer-bar-entity-row config=\\"[object Object]\\" hass=\\"[object Object]\\">
        <hui-generic-entity-row config=\\"[object Object]\\" no-secondary=\\"\\">
          <state-badge _showIcon=\\"true\\" class=\\"pointer\\" stateObj=\\"[object Object]\\" tabindex=\\"0\\">
            <ha-state-icon data-state=\\"on\\">…</ha-state-icon>
          </state-badge>
          <div class=\\"info pointer text-content\\" title=\\"switch\\">
            switch
          </div>
          <slot></slot>
          <div class=\\"bar-container\\">
            <div class=\\"bar\\">
              <div></div>
            </div>
          </div>
          <div class=\\"text-content\\">
            2
          </div>
        </hui-generic-entity-row>
      </timer-bar-entity-row>
    </div>
  </ha-card>
</timer-bar-card>
"
`;
